src/ipred_prepare.rs

```diff
@@ -245,8 +247,28 @@ pub fn rav1d_prepare_intra_edges<BD: BitDepth>(
         let left = &mut topleft_out[topleft_out_offset - sz..];
         if have_left {
             let px_have = cmp::min(sz, (h - y << 2) as usize);
-            for i in 0..px_have {
-                left[sz - 1 - i] = *(dst + (i as isize * stride - 1)).index::<BD>();
+            {
+                let slice_offset = if is_neg_stride {
+                    (px_have - 1) as isize * stride
+                } else {
+                    0
+                };
+                let dst_slice =
+                    &*(dst + slice_offset - 1isize).slice::<BD>((px_have - 1) * abs_stride + 1);
```

Comment:
The slice here overlaps ones from other threads, as the individual indices accessed are interleaved.  That's why `.index::<BD>()` works, but not a single `.slice::<BD>()`.  It's unfortunately a very tricky problem.